src/DebugEngineHost/RegistryMonitor.cs

```diff
@@ -0,0 +1,148 @@
+﻿// // Copyright (c) Microsoft. All rights reserved.
+// // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Remoting.Messaging;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using static System.Collections.Specialized.BitVector32;
+
+namespace Microsoft.DebugEngineHost
+{
+    internal class RegistryMonitor : IDisposable
+    {
+        #region Native Methods
+
+        /// <summary>
+        /// Filter for notifications reported by <see cref="RegistryMonitor"/>.
+        /// </summary>
+        [Flags]
+        public enum RegChangeNotifyFilter
+        {
+            /// <summary>Notify the caller if a subkey is added or deleted.</summary>
+            Key = 1,
+            /// <summary>Notify the caller of changes to the attributes of the key,
+            /// such as the security descriptor information.</summary>
+            Attribute = 2,
+            /// <summary>Notify the caller of changes to a value of the key. This can
+            /// include adding or deleting a value, or changing an existing value.</summary>
+            Value = 4,
+            /// <summary>Notify the caller of changes to the security descriptor
+            /// of the key.</summary>
+            Security = 8,
+        }
+
+        [DllImport("advapi32.dll", SetLastError = true)]
+        private static extern int RegOpenKeyEx(IntPtr hKey, string subKey, uint options, int samDesired,
+                                               out IntPtr phkResult);
```

Comment:
This wasn't used

Reply:
`Microsoft.Win32.RegistryKey.OpenSubKey` opens with `0x2001F` which covered `KEY_NOTIFY (0x0010)`